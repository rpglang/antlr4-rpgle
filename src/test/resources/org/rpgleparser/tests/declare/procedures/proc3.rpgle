       DCL-PR printf INT(10) ExtProc('Qp0zLprintf');
         OutputString POINTER Value OPTIONS(*STRING);
         *N POINTER Value OPTIONS(*STRING:*NOPASS);
         *N POINTER Value OPTIONS(*STRING:*NOPASS);
         // ....
       END-PR;
