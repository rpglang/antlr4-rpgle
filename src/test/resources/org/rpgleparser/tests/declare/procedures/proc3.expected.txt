/*===TOKENS===*/
DS_PrototypeStart        <DCL-PR>
ID                       <printf>
INT                      <INT>
'('                      <(>
NUMBER                   <10>
')'                      <)>
KEYWORD_EXTPROC          <ExtProc>
'('                      <(>
StringLiteralStart       <'>
StringContent            <Qp0zLprintf>
StringLiteralEnd         <'>
')'                      <)>
FREE_SEMI                <;>
ID                       <OutputString>
POINTER                  <POINTER>
KEYWORD_VALUE            <Value>
KEYWORD_OPTIONS          <OPTIONS>
'('                      <(>
ID                       <*STRING>
')'                      <)>
FREE_SEMI                <;>
MULT_NOSPACE             <*>
ID                       <N>
POINTER                  <POINTER>
KEYWORD_VALUE            <Value>
KEYWORD_OPTIONS          <OPTIONS>
'('                      <(>
ID                       <*STRING>
COLON                    <:>
ID                       <*NOPASS>
')'                      <)>
FREE_SEMI                <;>
MULT_NOSPACE             <*>
ID                       <N>
POINTER                  <POINTER>
KEYWORD_VALUE            <Value>
KEYWORD_OPTIONS          <OPTIONS>
'('                      <(>
ID                       <*STRING>
COLON                    <:>
ID                       <*NOPASS>
')'                      <)>
FREE_SEMI                <;>
HIDDEN:COMMENTS          <    //>
HIDDEN:COMMENTS_TEXT     <....>
DS_PrototypeEnd          <END-PR>
FREE_SEMI                <;>
/*===TREE===*/
(r
  (dcl_pr
    DCL-PR
    (identifier (free_identifier (idOrKeyword printf)))
    (datatype (datatypeName INT) (args ( (expression (number 10)) )))
    (keyword
      (keyword_extproc ExtProc ( (simpleExpression (literal ' Qp0zLprintf ')) ))
    )
    ;
    (dcl_pr_field
      (identifier (free_identifier (idOrKeyword OutputString)))
      (datatype (datatypeName POINTER))
      (keyword (keyword_value Value))
      (keyword
        (keyword_options
          OPTIONS
          (
          (identifier (free_identifier (idOrKeyword *STRING)))
          )
        )
      )
      ;
    )
    (dcl_pr_field
      (identifier (free_identifier * (idOrKeyword N)))
      (datatype (datatypeName POINTER))
      (keyword (keyword_value Value))
      (keyword
        (keyword_options
          OPTIONS
          (
          (identifier (free_identifier (idOrKeyword *STRING)))
          :
          (identifier (free_identifier (idOrKeyword *NOPASS)))
          )
        )
      )
      ;
    )
    (dcl_pr_field
      (identifier (free_identifier * (idOrKeyword N)))
      (datatype (datatypeName POINTER))
      (keyword (keyword_value Value))
      (keyword
        (keyword_options
          OPTIONS
          (
          (identifier (free_identifier (idOrKeyword *STRING)))
          :
          (identifier (free_identifier (idOrKeyword *NOPASS)))
          )
        )
      )
      ;
    )
    (end_dcl_pr END-PR)
    ;
  )
)
/*======*/